commit 29114a11a4e8f8a8e3f83376b4431714d10c4ec0
Author: Bob Peterson <rpeterso@redhat.com>
Date:   Fri Mar 15 12:28:56 2013 -0700

    fsck.gfs2: Check for interrupt when resolving duplicates
    
    This patch adds another check for interrupts while resolving duplicate
    block references in pass1b.
    
    rhbz#902920

diff --git a/gfs2/fsck/pass1b.c b/gfs2/fsck/pass1b.c
index 05f270e..d6b52e0 100644
--- a/gfs2/fsck/pass1b.c
+++ b/gfs2/fsck/pass1b.c
@@ -455,6 +455,9 @@ static int resolve_dup_references(struct gfs2_sbd *sdp, struct duptree *b,
 	int found_good_ref = 0;
 
 	osi_list_foreach_safe(tmp, ref_list, x) {
+		if (skip_this_pass || fsck_abort)
+			return FSCK_OK;
+
 		id = osi_list_entry(tmp, struct inode_with_dups, list);
 		dh->b = b;
 		dh->id = id;
